src/components/datasets/files/custom-actions-dialog/CustomActionsDialog.vue

```diff
@@ -0,0 +1,206 @@
+import BfDialogHeader from '../../../shared/bf-dialog-header/BfDialogHeader.vue'
+import DialogBody from '../../../shared/dialog-body/DialogBody.vue'
+import Request from '../../../../mixins/request/index'
+import EventBus from '../../../../utils/event-bus'
+
+import { mapGetters, mapState } from 'vuex'
+
+export default {
+  name: 'BfDeleteDialog',
+
+  components: {
+    BfDialogHeader,
+    DialogBody,
+    BfButton
+  },
+
+  mixins: [Request],
+
+  props: {
+    selectedFiles: {
+      type: Array
+    },
+    callingFromDeleted: {
+      type: Boolean,
+      default: false
+    },
+    selectedDeletedFiles: {
+      type: Array
+    }
+  },
+  data: function() {
+    return {
+      visible: false,
+      options: [],
+      value: ''
+    }
+  },
+
+  computed: {
+    ...mapGetters(['userToken', 'config', 'dataset', 'activeOrganization']),
+    ...mapState('integrationsModule', ['integrations']),
+    /**
+     * Checks for whether submit action should be allowed
+     * @returns {Boolean}
+     */
+    disableRunAction: function() {
+      return 'false'
+    },
+
+    /**
+     * Compute total files
+     * @return {Number}
+     */
+    totalFiles: function() {
+      return this.selectedFiles.length
+    },
+
+    /**
+     * Compute headline based on total files
+     * @return {String}
+     */
+    headline: function() {
+      return this.totalFiles > 1 ? 'items' : 'item'
+    },
+
+    /**
+     * Compute copy based on total files
+     * @return {String}
+     */
+    copy: function() {
+      return this.totalFiles > 1 ? 'these items' : 'this item'
+    }
+  },
+
+  watch: {
+    /**
+     * Watch file
+     */
+    file: function(file) {
+      if (Object.keys(file).length) {
+        this.packageForm.name = file.content.name
+      }
+    }
+  },
+
+  mounted() {
+    this.formatCustomIntegrationsOptions()
+  },
+
+  methods: {
+    /**
+     * Access integrations from global state and format options for input select
+     */
+    formatCustomIntegrationsOptions: function() {
+      this.options = this.integrations.map(integration => {
+        const [eventTargetType] = [...integration.eventTargets]
+        if (eventTargetType === 'CUSTOM') {
+          return {
+            value: integration.name,
+            label: integration.displayName
+          }
+        }
+      })
+    },
+
+    /**
+     * Closes the dialog
+     */
+    closeDialog: function() {
+      this.visible = false
+    },
+    /**
+     * Makes API Call to run custom event on target
+     */
+    runCustomEvent: function() {
+      // const fileIds = this.selectedFiles.map(item => item.content.id)
+      const url = `https://api.pennsieve.net/datasets/N:dataset:e2eaa76a-e0e0-4f25-97a8-6fa2bf6dd83f/event`
+      const message = JSON.stringify({
+        organizationId: `${this.activeOrganization.organization.id}`,
+        datasetId: `${this.dataset.content.id}`,
+        eventCategory: 'CATEGORY',
+        eventType: 'CUSTOM'
```

Comment:
Do we have an object / array with event types elsewhere that we can reference vs hard coding the type? Like an Events.CustomType.

Same thing for category

Reply:
@darrell-d Hi! Yep, we have that data. I'll make a note of this and when I go to enhance this feature in the future (we need to update it to support the 'FILE' or 'FILES' targets, for instance, we can update this so it reads the eventType directly from the integration data.